packages/plugins/Filter/src/store/index.ts

```diff
@@ -0,0 +1,189 @@
+    return
+  }
+  const [step, ...restPath] = path
+  if (!state[step]) {
+    Vue.set(state, step, {})
+  }
+  setState(state[step], restPath, value)
+}
+
+const parseTimeOption =
+  (timeDirection: 'last' | 'next') => (config: FilterConfigurationTimeOption) =>
+    config.amounts.map((amount) => ({
+      label: `common:plugins.filter.time.${timeDirection}.${config.unit}`,
+      component: null,
+      amount,
+      unit: config.unit,
+      now: null,
+      type: timeDirection,
+    }))
+
+const storeModule: PolarModule<FilterState, FilterGetters> = {
+  namespaced: true,
+  state: getInitialState(),
+  actions: {
+    setupModule({ getters: { filterConfiguration }, commit }): void {
+      Object.entries(filterConfiguration.layers).forEach(
+        ([layerId, { categories, time }]) => {
+          categories?.forEach?.(({ targetProperty, knownCategories }) => {
+            commit('setupState', {
+              path: ['category', layerId, targetProperty],
+              value: knownCategories.reduce((accumulator, current) => {
+                accumulator[current] = true
+                return accumulator
+              }, {}),
+            })
+          })
+          if (time) {
+            const { targetProperty, freeSelection, pattern } = time
+            commit('setupState', {
+              path: ['time', layerId],
+              value: {
+                targetProperty,
+                pattern: pattern || 'YYYY-MM-DD',
+                radioId: 0,
+                freeSelection: freeSelection ? [] : null,
+              },
+            })
+          }
+        }
+      )
+    },
+    toggleCategory({ getters, commit, dispatch }, payload) {
+      const value = !getters.getActiveCategory(payload)
+      commit('setCategory', { ...payload, value })
+      dispatch('updateFeatureVisibility', payload.layerId)
+    },
+    toggleCategoryAll({ getters, commit, dispatch }, payload) {
+      // 'indeterminate' to false intentionally (something had to be decided)
+      const value = !getters.getActiveCategoryAll(payload)
+      const { layerId } = payload
+      // @ts-expect-error | this call only happens if structures exist (generation in .vue)
+      getters
+        .getCategories(layerId)
+        .find(({ targetProperty }) => targetProperty === payload.targetProperty)
+        .knownCategories.forEach((knownCategory) => {
+          commit('setCategory', { ...payload, knownCategory, value })
+        })
+      dispatch('updateFeatureVisibility', payload.layerId)
+    },
+    changeTimeRadio({ commit, dispatch }, payload) {
+      commit('setTimeRadio', payload)
+      dispatch('updateFeatureVisibility', payload.layerId)
+    },
+    changeFreeSelection({ commit, dispatch }, { layerId, freeSelection }) {
+      commit('setFreeSelection', { layerId, freeSelection })
+      dispatch('updateFeatureVisibility', layerId)
+    },
+    updateFeatureVisibility({ state, rootGetters, getters }, layerId) {
+      updateFeatureVisibility({
```

Comment:
I find the duplicate naming a bit irritating. Is there a reason that the content of the util `updateFeatureVisibility` is not just placed here? Maybe I am also unsure about the line one needs to decide when writing actions or util-functions

Reply:
`utils/updateFeatureVisibility` is ~150 lines long and does not commit or dispatch any further, but only changes the map. I felt readability would increase if that logic was pushed to the utils folder. Now that I look at it, putting it right in the store would've broken the 300-lines-per-file rule, anyway.

I don't think there needs to be a hard criterium about where to put the code. This soft criterium suffices:

"If it fits, it sits."

![if-i-fits-i-sits-the-strict-and-adorable-code-of-cat-62-cute-pictures](https://github.com/Dataport/polar/assets/108349707/c4cc29b6-388e-4255-a2b2-e0f656c74eba)

Reply:
It would also fit as an action, written as another file 😉  
But I won't push on this too hard, if you got hard feelings 🎩